### PR TITLE
Do not load nor compile NIF on Unix

### PR DESCRIPTION
**Pull request by josevalim**

### CODE REVIEW

**Comment by josevalim** [comment]
I haven't tested on Windows, so I may have broken stuff. :D

File: lib/dll_loader_helper.ex [hunk 1]
```diff
@@ -3,14 +3,30 @@ defmodule DLLLoaderHelper do
...
+
   @doc """
   Add a directory to DLL search path for Windows.
   """
   @spec addDLLDirectory(binary()) :: :ok | {:error, String.t()}
   def addDLLDirectory(dir) when is_binary(dir) do
-    case :os.type() do
-      {:win32, _} -> :dll_loader_helper_nif.addDLLDirectory(dir)
-      _ -> :ok
-    end
```
Comment (josevalim) [inline lib/dll_loader_helper.ex:14]: If we don't load a NIF, then this version will stay in the module, which is exactly what we want for Unix! :)

File: lib/dll_loader_helper.ex [hunk 2]
```diff
@@ -3,14 +3,30 @@ defmodule DLLLoaderHelper do
...
+  @on_load :load_nif
+  def load_nif do
+    case :os.type() do
+      {:win32, _} ->
+        nif_file = '#{:code.priv_dir(:dll_loader_helper)}/dll_loader_helper'
+
+        case :erlang.load_nif(nif_file, 0) do
+          :ok -> :ok
+          {:error, {:reload, _}} -> :ok
+          {:error, reason} -> {:error, reason}
```
Comment (josevalim) [inline lib/dll_loader_helper.ex:15]: If you return not `:ok`, Erlang should already log it!

File: lib/dll_loader_helper.ex [hunk 3]
```diff
@@ -3,14 +3,30 @@ defmodule DLLLoaderHelper do
   DLL loader helper.
   """
 
+  require Logger
+
```
Comment (josevalim) [inline lib/dll_loader_helper.ex]:
```suggestion
```

File: mix.exs [hunk 4]
```diff
@@ -8,34 +8,23 @@ defmodule DllLoaderHelper.MixProject do
...
-      {:win32, _} -> "nmake"
-      _ -> "make"
-    end
-  end
-
-  def make_makefile() do
-    case :os.type() do
-      {:win32, _} -> "Makefile.win"
-      _ -> "Makefile.unix"
+      {:win32, _} -> [:elixir_make]
```
Comment (josevalim) [inline mix.exs:21]: This is the tricky to only run `elixir_make` on Windows. :) `nmake` and `Makefile.win` are the defaults then!

**Comment by cocoa-xu** [comment]
This looks good to me! I haven't setup CI for this repo, but I'll check it on my local machine. There seems to be no issues.

**Comment by cocoa-xu** [comment]
It's now okay to go. Thanks for the PR! 😄

**Pull request merged**